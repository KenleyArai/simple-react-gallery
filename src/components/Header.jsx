import React from "react";
import Lazyload from "react-lazyload";
import PropTypes from "prop-types";

import HeaderStyle from "styled/HeaderStyle";

const Header = ({ children }) => {
  return <HeaderStyle>{children}</HeaderStyle>;
};

Header.propTypes = {
  children: PropTypes.string
};

Header.defaultProps = {
  children: ""
};

export default Header;
