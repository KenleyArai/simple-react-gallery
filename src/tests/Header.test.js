import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import "jest-styled-components";
configure({ adapter: new Adapter() });

import Header from "components/Header";

describe("Header", () => {
  it("should render without children", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with children", () => {
    const wrapper = shallow(<Header>Test Header</Header>);
    expect(wrapper).toMatchSnapshot();
  });
});
