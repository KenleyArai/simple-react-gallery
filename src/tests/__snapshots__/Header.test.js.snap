// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render with children 1`] = `
<styled.div
  theme={
    Object {
      "primary": "#fff",
      "secondary": "",
      "tertiary": "",
    }
  }
>
  <styled.span>
    Test Header
  </styled.span>
</styled.div>
`;

exports[`Header should render without children 1`] = `
<styled.div
  theme={
    Object {
      "primary": "#fff",
      "secondary": "",
      "tertiary": "",
    }
  }
>
  <styled.span />
</styled.div>
`;
